src/main/java/com/MTAPizza/Sympoll/pollmanagementservice/validator/Validator.java

```diff
@@ -117,8 +117,11 @@ private void validateGroupIdExist(String groupId) {
         }
     }
 
-    private void validateUserHasPermissions(UUID userId, String groupId) {
-        // TODO: change this validation to send request to group service
+    private void validateUserHasPermissions(UUID userId, String groupId, UUID pollId) {
+        if(!isUserCreatedThePoll(userId,pollId) && !isUserGroupModeratorOrAdmin(userId, groupId)) {
+            log.warn("User {} has no permissions to delete the poll", userId);
+            throw new IllegalArgumentException("User " + userId + " has no permissions to delete the poll");
```

Comment:
Should throw a AccessDeniedException or a custom exception, to better explain the issue in the action that was performed.